### PR TITLE
Use RSA CRT parameters in FIPS self tests.

### LINKED ISSUE #19488: FIPS self-test data for RSA-CRT contains incorrect parameters
At https://github.com/openssl/openssl/blob/master/providers/fips/self_test_data.inc#L1269-L1279:
```
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_N, rsa_n),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_E, rsa_e),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_D, rsa_d),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_FACTOR, rsa_p),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_FACTOR, rsa_q),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_EXPONENT, rsa_dp),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_EXPONENT, rsa_dq),
    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_COEFFICIENT, rsa_qInv),
    ST_KAT_PARAM_END()
```
However, https://github.com/openssl/openssl/blob/master/providers/implementations/keymgmt/rsa_kmgmt.c#L244-L252 defines the correct parameters (in FIPS mode, which is relevant for the FIPS self-test data) as:
```
# define RSA_KEY_MP_TYPES()                                                    \
OSSL_PARAM_BN(OSSL_PKEY_PARAM_RSA_FACTOR1, NULL, 0),                           \
OSSL_PARAM_BN(OSSL_PKEY_PARAM_RSA_FACTOR2, NULL, 0),                           \
OSSL_PARAM_BN(OSSL_PKEY_PARAM_RSA_EXPONENT1, NULL, 0),                         \
OSSL_PARAM_BN(OSSL_PKEY_PARAM_RSA_EXPONENT2, NULL, 0),                         \
OSSL_PARAM_BN(OSSL_PKEY_PARAM_RSA_COEFFICIENT1, NULL, 0),
```
See also https://github.com/openssl/openssl/blob/master/crypto/rsa/rsa_mp_names.c

As a result of this bug, the actual keys loaded will never be RSA-CRT keys.

### PR DESCRIPTION
**Pull request by slontis**
Fixes #19488

Use the correct OSSL_PKEY_PARAM_RSA CRT names fior the self tests. The invalid names cause CRT parameters to be silently ignored.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
The OTC decision was to just document the problem for now, rather that cause either a compile or runtime error by either  - removing the defines, or aliasing to valid values.
The defines will be removed in the next major release.

File: CHANGES.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ OpenSSL 3.2
 
 ### Changes between 3.0 and 3.2 [xx XXX xxxx]
 
+ * Removed all references to invalid OSSL_PKEY_PARAM_RSA names for CRT parameters
+   in OpenSSL code.
+   Applications should not use the names OSSL_PKEY_PARAM_RSA_FACTOR,
+   OSSL_PKEY_PARAM_RSA_EXPONENT and OSSL_PKEY_PARAM_RSA_COEFFICIENT.
+   Use the numbered names such as OSSL_PKEY_PARAM_RSA_FACTOR1 instead.
+   Using these invalid names may cause algorithms to use slower methods
+   that ignore the CRT parameters.
```
Comment (paulidale) [inline CHANGES.md:33]: Shouldn't something a bit stronger be here?

It will ignore everything you give it.  Multiprime won't be. ...

**Review by t-j-h — APPROVED** [review]

**Review by levitte — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (t8m) [reply CHANGES.md:33]: Hmm, also the changes entry should be in 3.0 section as this is certainly going there too. We should aim for having it in 3.0.7.

**Review by t8m — APPROVED** [review]
Approving. I'll move the changes entry on the master branch in a subsequent PR.

**Comment by t8m** [comment]
Merged to master, 3.1 and 3.0 branches. The CHANGES.md entry was moved to the right place on cherry-pick.

**Pull request closed without merging**